addons/hr_expense/models/hr_expense.py

```diff
@@ -706,7 +706,13 @@ def message_new(self, msg_dict, custom_values=None):
             '|',
             ('work_email', 'ilike', email_address),
             ('user_id.email', 'ilike', email_address)
-        ], limit=1)
+        ])
+
+        for emp in employee:
+            if emp.company_id == emp.user_id.company_id:
+                employee = emp
+                break
+            employee = employee[0]
```

Comment:
```suggestion
        ]).filtered(lambda e: e.company_id == e.user_id.company_id)
```
A bit easier to read/understand IMO :wink: